numpy/core/src/multiarray/lowlevel_strided_loops.c.src

```diff
@@ -1352,17 +1401,16 @@ PyArray_TransferMaskedStridedToNDim(npy_intp ndim,
 
             /* A loop for dimensions 0 and 1 */
             for (i = 0; i < shape1; ++i) {
+                args[0] = src; args[1] = dst;
                 if (shape0 >= count) {
-                    return stransfer(
-                            dst, dst_stride0, src, src_stride,
-                            mask, mask_stride,
-                            count, src_itemsize, data);
+                    return stransfer(&cast_info->context,
+                            args, &count, strides, mask,
+                            mask_stride, cast_info->auxdata);
```

Comment:
this seems like a false negative, surely we are hitting this code path?

Reply:
Hmmm, probably not a false positive... These are masked buffered casts that are only used in `NpyIter` so they are pretty darn specific and little used. (And this branch is probably specific to large arrays, have to think a bit how to test it right.)

Reply:
I added/expanded a test... Its not pretty, and confusing though :/.